python/paddle/fluid/framework.py

```diff
@@ -1866,6 +1866,35 @@ def __setitem__(self, item, value):
         starts = []
         ends = []
         max_integer = sys.maxsize
+
+        def replace_ellipsis(item):
+            # Use slice(None) to replace Ellipsis.
+            # For var, var.shape = [3,4,5,6]
+            #
+            #   var[..., 1:2] -> var[:, :, :, 1:2]
+            #   var[0, ...] -> var[0]
+            #   var[0, ..., 1:2] -> var[0, :, :, 1:2]
+
+            item = list(item)
```

Comment:
I am not sure about the purpose of line 1878:
Is `item` always a list? If yes, you don't have to do this. If not, item = replace_ellipsis(item) would change the type of `item`, which is different to past code. Will this cause some problems?

Reply:
`item` is a tuple and it can't be modified. In order to modify the value of `item`, and call the metheds `count()` and `index()` of `list`, here converts it to `list`. 

> If not, item = replace_ellipsis(item) would change the type of item, which is different to past code. Will this cause some problems?

No. Here, it's ok that `item` is a list after calling `replace_ellipsis`.